loopchain/consensus/syncer.py

```diff
@@ -0,0 +1,160 @@
+        if gap < RAISE_EVENT_INTERVAL:
+            return
+
+        goal = min(conf.CITIZEN_ASYNC_RESULT_MAX_SIZE+1, self._max_height_in_nodes-RAISE_EVENT_INTERVAL)
+        for i in range(1, goal):
+            height = self.__blockchain.block_height + i
+            try:
+                if height in self._request_history_list.keys():
+                    during_request_time = time.time()-self._request_history_list[height][1]
+                    if during_request_time > conf.LFT_SYNC_REQUEST_WAIT:
+                        retry_target_index = (self._request_history_list[height][0] + 1) % len(self._stub_list)
+                        _request(retry_target_index, height)
+                        self._request_history_list[height] = [retry_target_index, time.time()]
+                else:
+                    _request(self._target_index, height)
+                    self._request_history_list[height] = [self._target_index, time.time()]
+                    self._target_index = (self._target_index+1) % len(self._stub_list)
+
+            except IndexError:
+                self._target_index = 0
+
+            await asyncio.sleep(0)
+
+    def receive_vote(self, vote: 'BlockVote'):
+        height = vote.block_height
+        self._max_height_in_nodes = max(height, self._max_height_in_nodes)
+
+        if abs(self.__blockchain.block_height-height) < conf.CITIZEN_ASYNC_RESULT_MAX_SIZE:
+            if height not in self._vote_info_other_nodes.keys():
+                self._vote_info_other_nodes[height] = []
+
+            self._vote_info_other_nodes[height].append(vote)
+
+    def receive_data(self, block_data: 'Block'):
+        height = block_data.header.height
+        self._max_height_in_nodes = max(height, self._max_height_in_nodes)
+
+        diff_height_info = abs(self.__blockchain.block_height-self._max_height_in_nodes)
+        if diff_height_info < conf.CITIZEN_ASYNC_RESULT_MAX_SIZE:
+            if height not in self._data_info_other_nodes.keys():
+                self._data_info_other_nodes[height] = []
+
+            self._data_info_other_nodes[height].append(block_data)
+
+            # Check Sync mode
+            # If height make different 3 value between to written block height and to received block height,
+            # It information is to responded for Synchronize.
+            # So It need to raise to divide Block information and Vote Information.
+            if RAISE_EVENT_INTERVAL < diff_height_info:
+                for vote in block_data.prev_votes:
+                    self.receive_vote(vote)
+
+    async def _raise_event(self):
+        def __raise_vote(height: int):
+            while self._vote_info_other_nodes[height]:
+                vote_info = self._vote_info_other_nodes[height].pop()
+                event = ReceiveVoteEvent(vote_info)
+                self._event_system.simulator.raise_event(event)
+
+            del self._vote_info_other_nodes[height]
```

Comment:
Any reason to re-define this method in every call, which uses `self` property?

Reply:
It work on normal case and `NONE_VOTE_HEIGHT`. So, It written function.
Function is move at Class member function `def _raise_vote(self, height: int)`